server/resp/src/main/java/org/infinispan/server/resp/commands/ArgumentUtils.java

```diff
@@ -12,7 +10,29 @@ public final class ArgumentUtils {
    private ArgumentUtils() {
 
    }
+
    public static long toLong(byte[] argument) {
-      return Long.parseLong(new String(argument, StandardCharsets.US_ASCII));
+      if (argument == null || argument.length == 0)
+         throw new NumberFormatException("Empty argument");
+
+      boolean negative = false;
+      int i = 0;
+      if (argument[0] < '0') {
+         if (argument[0] != '-')
```

Comment:
Seems that `-` is returned as 0.
to avoid it
```suggestion
         if (argument[0] != '-' || argument.length==1)
```